### ISSUE TITLE
Direct Download to Ceph Primary storage fails

### ISSUE THREAD

**Issue by wido** [apache/cloudstack#3065]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
KVM
Direct Download
~~~

##### CLOUDSTACK VERSION
~~~
master / 4.12
~~~

##### CONFIGURATION
Trying to deploy a direct download template on Ceph / RBD.

##### OS / ENVIRONMENT
- Ubuntu 18.04
- Ceph Luminous 12.2.8

##### SUMMARY
Deployment of the template fails on RBD.

##### STEPS TO REPRODUCE
Try to deploy a Direct Download template on RBD shared storage and you will see it fails.

##### EXPECTED RESULTS
~~~
deployment should work
~~~

**Comment by rhtyd** [comment]
The direct download feature needs to be supported for ceph cc @nvazquez

**Comment by nvazquez** [comment]
@wido @rhtyd currently NFS is supported, needs to be extended for Ceph

**Comment by nvazquez** [comment]
Moved to the next major milestone

**Comment by andrijapanicsb** [comment]
Do we have some work in progress around improving direct downloads @nvazquez - I recall we discussed this at some point for various storage types (Ceph and SolidFire included)?

**Comment by nvazquez** [comment]
Yes @andrijapanicsb but the improvement will not include managed storage atm. This will need to be moved to the next release/milestone

**Comment by rhtyd** [comment]
I think we added support for direct download agnostic of storage cc @nvazquez @shwstppr ? Is this still an issue then?

**Comment by andrijapanicsb** [comment]
We don't support Ceph atm, the framework is storage-agnostics, but the implementation for Ceph/SolidFire is not done.

**Comment by nvazquez** [comment]
I've renamed the title and added this to the GSOC 2023 ideas list

**Comment by DaanHoogland** [comment]
@wido, is this still an issue?

**Comment by wido** [comment]
> [@wido](https://github.com/wido), is this still an issue?

I think it is, but I haven't done any testing nor do I have the ability at the moment.

**Comment by nvazquez** [comment]
Hi @wido I have created this PR to add support for Direct downloads: https://github.com/apache/cloudstack/pull/11069, can you please review?
